crates/chain/src/mem_pool.rs

```diff
@@ -25,6 +22,7 @@ const MAX_DATA_BYTES_LIMIT: usize = 25_000;
 
 /// TODO remove txs from pool if a new block already contains txs
 pub struct MemPool {
+    store: Store,
```

Comment:
The mem pool should always use the overlay store. otherwise, the state may be inconsistent.